dashboard/src/pages/settings.tsx

```diff
@@ -0,0 +1,223 @@
+    gap: '16px',
+
+    background: vars.bg.white3,
+    border: `1px solid ${vars.bg.white4}`,
+    borderRadius: '4px',
+    padding: '8px 12px',
+  },
+})
+
+const PublicKey = styled('div', {
+  base: {
+    background: vars.bg.white1,
+    border: `1px solid ${vars.bg.white5}`,
+    borderRadius: '4px',
+    padding: '8px 12px',
+  },
+})
+
+const PublicKeyContainer = styled('div', {
+  base: {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '6px',
+
+    background: vars.bg.white2,
+    border: `1px solid ${vars.bg.white5}`,
+    borderRadius: '4px',
+    padding: '8px 12px',
+  },
+})
+
+const CreateKeyContainer = styled('div', {
+  base: {
+    background: vars.bg.white3,
+    border: `1px solid ${vars.bg.white4}`,
+    borderRadius: '4px',
+    padding: '8px 12px',
+  },
+})
+
+const CreatingKeyContainer = styled('form', {
+  base: {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '16px',
+
+    background: vars.bg.white2,
+    border: `1px solid ${vars.bg.white5}`,
+    borderRadius: '4px',
+    padding: '8px 12px',
+  },
+})
+
+export const FormButton = styled('div', {
+  base: {
+    marginLeft: '4px',
+  },
+})
+
+export default () => {
+  const params = useParams()
+  const [userKeys, { refetch: refetchApp }] = createResource(() => client.getUserKeys({}))
+
+  const [createKeyToggle, setCreateKeyToggle] = createSignal(false)
+  const [createKey, setCreateKey] = createStore(new CreateUserKeyRequest())
+
+  const [deleteKeyID, setDeleteKeyID] = createStore(new DeleteUserKeyRequest())
+
+  let formContainer: HTMLFormElement
+  const createKeyRequest: JSX.EventHandler<HTMLButtonElement, MouseEvent> = async (e) => {
+    // prevent default form submit (reload page)
+    e.preventDefault()
+
+    // validate form
+    if (!formContainer.reportValidity()) {
+      return
+    }
+
+    try {
+      const res = await client.createUserKey(createKey)
+      toast.success('User Key を登録しました')
+      refetchApp()
+    } catch (e) {
+      console.error(e)
+      // gRPCエラー
+      if (e instanceof ConnectError) {
+        toast.error('User Key の登録に失敗しました\n' + e.message)
+      }
+    }
+  }
+
+  const deleteKeyRequest: JSX.EventHandler<HTMLButtonElement, MouseEvent> = async (e) => {
+    // prevent default form submit (reload page)
+    e.preventDefault()
+
+    try {
+      await client.deleteUserKey(deleteKeyID)
+      toast.success('User Key を削除しました')
+      refetchApp()
+    } catch (e) {
+      console.error(e)
+      // gRPCエラー
+      if (e instanceof ConnectError) {
+        toast.error('User Key の削除に失敗しました\n' + e.message)
+      }
+    }
+  }
+
+  return (
+    <Container>
+      <Header />
+      <PageTitle>User Settings</PageTitle>
+      <ContentContainer>
+        <MainContentContainer>
+          <UserKeysContainer>
+            <SidebarTitle>登録済みSSH公開鍵</SidebarTitle>
+            if (!userKeys()) return
+            return userKeys().keys
+            <For each={!userKeys() || userKeys().keys}>
```

Comment:
```suggestion
            <For each={userKeys()?.keys}>
```

上記のようにオプショナルチェーンを使うと、`userKeys()`が読み込み中でundefinedの場合を考慮しつつシンプルに書くことができると思います。

[オプショナルチェーン (?.) - JavaScript | MDN](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

Reply:
もっと外側を <Show when={userKeys()}> で囲って、読み込み中の場合は表示しないようにしてもよいかもしれませんね

Reply:
何かエラーが起きて表示されなかったとき、`登録済みSSH公開鍵`とだけ表示されてたら何も登録されてないと勘違いされることがあると思ったので、もっと外側で囲むのはアリだと思ったのですが、やってみたら一瞬だけチカッとするような感じになってしまったので、Showの位置は同じところで、読み込まれてないときは読み込み中と表示するようにしました。